lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -71,6 +76,8 @@ def render(engine, data, *)
             end
           end
 
+          app.use RequestSpanMiddleware
```

Comment:
Good questions.
I've introduced the `RequestSpanMiddleware` because its actually impossible to fetch all response headers from `app.after` hook, as it this filter happens before the response is fully rendered.

As such I think its still worth to have Sinatra specific instrumentation in `Tracer` as it provides access to Sinatra's specific code and information, which might not be possible from within Rack Middleware.

One thing that could be improved here though is coupling via calling order. I.e. Sinatra::Tracer relies on the span to be created with `RequestSpanMiddleware`. While in general this is not a problem because Middleware is always executed before. However its still unnecessary coupling I'll address that in next commit